packages/block-library/src/navigation/edit/index.js

```diff
@@ -693,48 +693,55 @@ function Navigation( {
 	const isManageMenusButtonDisabled =
 		! hasManagePermissions || ! hasResolvedNavigationMenus;
 
+	const MenuInspectorControls = ( { currentMenuId = null } ) => (
+		<InspectorControls>
+			<PanelBody
+				title={
+					isOffCanvasNavigationEditorEnabled ? null : __( 'Menu' )
+				}
+			>
+				{ isOffCanvasNavigationEditorEnabled ? (
+					<>
+						<HStack className="wp-block-navigation-off-canvas-editor__header">
+							<Heading
+								className="wp-block-navigation-off-canvas-editor__title"
+								level={ 2 }
+							>
+								{ __( 'Menu' ) }
+							</Heading>
+							<WrappedNavigationMenuSelector
+								currentMenuId={ currentMenuId }
+							/>
+						</HStack>
+						{ currentMenuId && isNavigationMenuMissing ? (
+							<p>{ __( 'Select or create a menu' ) }</p>
+						) : (
+							<OffCanvasEditor
+								blocks={ innerBlocks }
+								isExpanded={ true }
+								selectBlockInCanvas={ false }
+							/>
+						) }
+					</>
+				) : (
+					<>
+						<WrappedNavigationMenuSelector
+							currentMenuId={ currentMenuId }
+						/>
+						<ManageMenusButton
+							disabled={ isManageMenusButtonDisabled }
```

Comment:
what do you mean by encapsulated inside the component?

Reply:
I mean you're relying on variables defined outside the component scope. If this component were inside it's own file you couldn't do that. Passing that data via props would be the alternative and makes the component easier to...

> reason about
> refactor later into a separate file

Reply:
Ah yeah, that would be a good next step :)

Reply:
Yes agreed. I suppose that's why I'm suggesting doing it now (and every time we refactor to a component) because it makes it easier to take the next step.